source/Ship.cpp

```diff
@@ -1104,21 +1105,27 @@ vector<string> Ship::FlightCheck() const
 	// Report the first error condition that will prevent takeoff:
 	if(IdleHeat() >= MaximumHeat())
 		checks.emplace_back("overheating!");
-	else if(energy <= 0.)
+	else if(energy <= 0. && !docked)
```

Comment:
We can only check for docked once, and then keep the logic as it was before, in this intended block.
Like this:
```cpp
	else if(!docked)
	{
		if(energy <= 0.)
			checks.emplace_back("no energy!");
		else if((energy - consuming <= 0.) && (fuel <= 0.))
			checks.emplace_back("no fuel!");
		else if(!thrust && !reverseThrust && !afterburner)
			checks.emplace_back("no thruster!");
		else if(!turn)
			checks.emplace_back("no steering!");
	}
```